server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -90,16 +91,23 @@ class InteractiveSession(
     }
     builder.setConf(RSCConf.Entry.SESSION_KIND.key, kind.toString)
 
-    val allJars = livyJars(livyConf) ++ request.jars
+    val allJars = ArrayBuffer((livyJars(livyConf) ++ request.jars): _*)
```

Comment:
This code will change a lot when I push an outstanding PR, so you'll have conflicts to resove; while doing that I'd encourage you to try to avoid mutable data structures, they tend to make the code less readable.